File: python/0/2000/2448.py
import sys
input = lambda: sys.stdin.readline().rstrip()

tri1 = [
    "  *  ",
    " * * ",
    "*****"
]

n = int(input())

out = [[' ' for _ in range(2 * n - 1)] for _ in range(n)]
n //= 3

def tri(n,x,y):
    global out

    if n == 1:
        for i in range(3):
            for j in range(5):
               out[y + i][x + j] = tri1[i][j]
    else:
        tri(n//2,x,y+3*n//2)
        tri(n//2,x+3*n//2,y)
        tri(n//2,x+3*n,y+3*n//2)

tri(n,0,0)
for wo in out:
    print(*wo,sep="")

File: Rust/0/2000/2448.rs
use std::{fmt::Write, io::{self, BufRead}};

fn input() -> String {
    let stdin = io::stdin();
    let mut line = String::new();
    stdin.lock().read_line(&mut line).unwrap();
    line.trim().to_string()
}

fn tri(n:i32) -> Vec<String> {
    if n == 3 {
        vec!["  *  ".to_string()," * * ".to_string(),"*****".to_string()]
    } else {
        let last = tri(n/2);
        let mut out = vec![];
        let step = " ".repeat((n/2) as usize);
        
        for wo in &last {
            out.push(format!("{step}{wo}{step}"));
        }
        for wo in last {
            out.push(format!("{wo} {wo}"));
        }
    
        out
    }
}

fn main() {
    let mut res = String::new();
    let t = input().parse().unwrap();
    let out = tri(t);
    for wo in out {
        writeln!(res,"{wo}").unwrap()
    }
    print!("{res}")
}

// 제출 번호 : 90077015, 메모리 : 70344, 시간 : 52